### DISCUSSION TITLE
geometry don't move picture

### DISCUSSION THREAD

**Discussion by nicolasdekaise** [ImageMagick/ImageMagick#5114] [Help]
Hi everyone,

I'm trying to place a picture (scenar-021.psd) over another (flatBg.psd) and place it to a precise position, but scenar-021.psd is always stucked in the top right corner

I'm working in command line under linux

`convert -size 2029x2000 img/flatBg.psd \( -geometry 198x102+915+289 img/scenar-021.psd \) -flatten output3.psd`

Any idea ?
Thank you !

**Comment by snibgo** [comment]
"-flatten" is shorthand for "-layers flatten".

For "-layers, "-geometry" is not relevant. Instead, "-page" is used.

"-geometry" is used for "-composite".

**Reply by nicolasdekaise** [reply]
Thank you ! So it work, sort of, but how do I use gravity with "-page" ?

**Comment by snibgo** [comment]
"-layers flatten" is not sensitive to "-gravity". "-layers composite" is sensitive to gravity, but you need to have a "null:" image between the two input images, like this:
```
magick wizard: null: rose: -gravity south -layers composite x.png
```

See https://imagemagick.org/script/command-line-options.php#layers

An alternative if you use "magick" is to do calculations within the "-page", like this:
```
magick wizard: -set option:WW %w ( rose: -page +%[fx:WW-w]+150 ) -layers flatten x.png
```
Note that calcs within "-page" won't work in IM v6, or IM v7 "magick convert".
